In [1]:
import pandas as pd

### Dataset info
| show_id | type | title | director | cast | country | date_added | release_year | rating | duration | listed_in | description |
| :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| 8807 unique values| Movie 70%<br> TvShow 30% | 8807 unique values | [null] 30%<br>Other 70% | [null] 9%<br> Other 90% | country of prodcution | date it was added on netflix | Actual release year of the movie/show | Tv-rating | Total duration in minutes or number of seasons | Genre<br> (Dramas, Documentaries, etc) | The summary description |

In [ ]:
# Read the data
movies = pd.read_csv("../data/netflix_titles.csv")

# We keep just the first 1000 samples
movies = movies[:1000]

print(f"Number of samples: {len(movies)}")

Number of samples: 1000


In [3]:
movies.head(5)

,show_id,type,title,director,cast,country,date_added,release_year,rating,duration,listed_in,description
0,s1,Movie,Dick Johnson Is Dead,Kirsten Johnson,NaN,United States,"September 25, 2021",2020,PG-13,90 min,Documentaries,"As her father nears the end of his life, filmm..."
1,s2,TV Show,Blood & Water,NaN,"Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...",South Africa,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, TV Dramas, TV Mysteries","After crossing paths at a party, a Cape Town t..."
2,s3,TV Show,Ganglands,Julien Leclercq,"Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...",NaN,"September 24, 2021",2021,TV-MA,1 Season,"Crime TV Shows, International TV Shows, TV Act...",To protect his family from a powerful drug lor...
3,s4,TV Show,Jailbirds New Orleans,NaN,NaN,NaN,"September 24, 2021",2021,TV-MA,1 Season,"Docuseries, Reality TV","Feuds, flirtations and toilet talk go down amo..."
4,s5,TV Show,Kota Factory,NaN,"Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...",India,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, Romantic TV Shows, TV ...",In a city of coaching centers known to train I...


In [4]:
# Auxilary function to count the number of occurrences 
# of the values for a given attribute

def counter(attr):
    count = {}

    if attr not in movies.columns:
        return "Error: attribute not present"
    
    col = movies[attr]

    for elem in col:
        if elem in count:
            count[elem] += 1
        else:
            count[elem] = 1

    return count

In [6]:
count = counter('rating')
print(count)

{'PG-13': 101, 'TV-MA': 339, 'PG': 44, 'TV-14': 211, 'TV-PG': 64, 'TV-Y': 35, 'TV-Y7': 59, 'R': 120, 'TV-G': 25, 'G': 2}


### Cassandra CQL files

In the following we prepare the cql files to create and populate the database.

#### Database *"schema"*

``` cql
CREATE TABLE Netflix_titles (  
    show_id text,
    type text, 
    title text, 
    rating text,   
    duration text, 
    director text, 
    cast set<text>,     ← NOTE: Cassandra is able to efficiently handle also collections
    release_year int,
    country text, 
    PRIMARY KEY (type, show_id)  
) 
```

We are also going to define a partition based on the ratings.
`{'PG-13': 101, 'TV-MA': 339, 'PG': 44, 'TV-14': 211, 'TV-PG': 64, 'TV-Y': 35, 'TV-Y7': 59, 'R': 120, 'TV-G': 25, 'G': 2}`

```cql
CREATE TABLE ratings (
    show_id text, 
    rating text, 
    type text, 
    title text, 
    duration text, 
    description text, 
    PRIMARY KEY (rating, type, show_id)
)
```

In [ ]:
def generate_cql_files(data):
    ''' 
        data: DataFrame
    '''

    num_rows = len(data)

    for idx in range(num_rows):
        row = data.iloc[idx]

        



In [32]:
rows = len(movies)

for idx in range(rows): 
    print(len(movies.iloc[idx]))
    break

#print(movies.shape)

12
